src/full/Agda/Interaction/Options.hs

```diff
@@ -298,6 +300,8 @@ checkOpts :: Flag CommandLineOptions
 checkOpts opts
   | not (matches [optGHCiInteraction, isJust . optInputFile] <= 1) =
       throwError "Choose at most one: input file or --interaction.\n"
+  | not (matches [optJSONInteraction, isJust . optInputFile] <= 1) =
+      throwError "Choose at most one: input file or --interaction-json.\n"
   | or [ p opts && matches ps > 1 | (p, ps) <- exclusive ] =
```

Comment:
I'm guessing it does not make sense to use `interaction` and
`interaction-json` together either. So instead I would write
only one guard for the three possibilities:

```haskell
not (matches [optGHCiInteraction, optJSONInteraction, isJust . optInputFile] <= 1) =
  throwError "Choose at most one: input file, --interactive, or --interaction-json.\n"
```